runtime/parachains/src/paras_inherent/weights.rs

```diff
@@ -110,13 +108,26 @@ pub fn signed_bitfields_weight<T: Config>(bitfields_len: usize) -> Weight {
 pub fn backed_candidate_weight<T: frame_system::Config + Config>(
 	candidate: &BackedCandidate<T::Hash>,
 ) -> Weight {
-	if candidate.candidate.commitments.new_validation_code.is_some() {
-		<<T as Config>::WeightInfo as WeightInfo>::enter_backed_candidate_code_upgrade()
-	} else {
-		<<T as Config>::WeightInfo as WeightInfo>::enter_backed_candidates_variable(
-			candidate.validity_votes.len() as u32,
-		)
-	}
+	let v = candidate.validity_votes.len() as u32;
+	let ump = candidate
+		.candidate
+		.commitments
+		.upward_messages
+		.iter()
+		.map(|m| m.len() as u32)
+		.sum();
```

Comment:
a comment why it can't overflow would be nice
I guess because we can't allocate that much in WASM (4*10**9)